### PR TITLE
Running All the CI workflows on Release branches along with main branch

### LINKED ISSUE #7434: Github Actions: Run All the CI workflows on Release branches
All the Github Actions the Repository Houses, should be running with changes made on Release Branches that follows a pattern of `0.X.X`.

### PR DESCRIPTION
**Pull request by Ashok810**
### Describe your changes :

fixes #7434 

#
### Type of change :

- [x] Improvement
- [x] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Frontend Preview (Screenshots) :
<p align="center">For frontend related change, please link screenshots of your changes preview! Optional for backend related changes.
</p>

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] All new and existing tests passed.

#

### CODE REVIEW

File: .github/workflows/sync-docs.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@
 name: Sync publish docs
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/sync-docs.yml:16]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/selenium-noIngestion-tests.yml [hunk 2]
```diff
@@ -16,12 +16,12 @@ name: Selenium No Ingestion Java CI
...
 on:
   push:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
     paths:
       - 'openmetadata-ui/src/main/resources/ui/**'
       - 'openmetadata-ui/src/test/java/org/openmetadata/catalog/selenium/pagesWithoutData/**'
   pull_request:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
```
Comment (akash-jain-10) [inline .github/workflows/selenium-noIngestion-tests.yml:24]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/selenium-noIngestion-tests.yml [hunk 3]
```diff
@@ -16,12 +16,12 @@ name: Selenium No Ingestion Java CI
 
 on:
   push:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
```
Comment (akash-jain-10) [inline .github/workflows/selenium-noIngestion-tests.yml:19]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/py-tests.yml [hunk 4]
```diff
@@ -12,12 +12,12 @@
...
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
     paths:
       - ingestion/**
       - openmetadata-service/**
   pull_request_target:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/py-tests.yml:20]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/py-tests.yml [hunk 5]
```diff
@@ -12,12 +12,12 @@
 name: py-tests
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/py-tests.yml:15]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/py-checkstyle.yml [hunk 6]
```diff
@@ -15,7 +15,7 @@ name: Python Checkstyle
 # access to secrets
 on:
   pull_request_target:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
```
Comment (akash-jain-10) [inline .github/workflows/py-checkstyle.yml:18]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/openmetadata-ingestion-core-version-comment.yml [hunk 7]
```diff
@@ -15,7 +15,7 @@ name: OpenMetadata Ingestion Core Version Check Comment
 # access to secrets
 on:
   pull_request_target:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
```
Comment (akash-jain-10) [inline .github/workflows/openmetadata-ingestion-core-version-comment.yml:18]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/maven-build.yml [hunk 8]
```diff
@@ -13,12 +13,12 @@ name: Maven CI
...
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
     paths:
       - "openmetadata-service/**"
       - "openmetadata-ui/**"
   pull_request_target:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/maven-build.yml:21]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/maven-build.yml [hunk 9]
```diff
@@ -13,12 +13,12 @@ name: Maven CI
 
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/maven-build.yml:16]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/java-checkstyle.yml [hunk 10]
```diff
@@ -15,7 +15,7 @@ name: Java Checkstyle
 # access to secrets
 on:
   pull_request_target:
-    branches: [ main ]
+    branches: [ main, '0.[0-9]+.[0-9]+' ]
```
Comment (akash-jain-10) [inline .github/workflows/java-checkstyle.yml:18]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/docs-tests.yml [hunk 11]
```diff
@@ -13,7 +13,7 @@
 name: docs-tests
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/docs-tests.yml:16]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/cypress-integration-tests-postgresql.yml [hunk 12]
```diff
@@ -16,7 +16,7 @@ name: PostgreSQL Cypress Integration Tests
 
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-postgresql.yml:19]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/cypress-integration-tests-mysql.yml [hunk 13]
```diff
@@ -16,7 +16,7 @@ name: MySQL Cypress Integration Tests
 
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/cypress-integration-tests-mysql.yml:19]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

File: .github/workflows/airflow-apis-tests-3_9.yml [hunk 14]
```diff
@@ -13,7 +13,7 @@
 name: airflow-apis-tests
 on:
   push:
-    branches: [main]
+    branches: [main, '0.[0-9]+.[0-9]+']
```
Comment (akash-jain-10) [inline .github/workflows/airflow-apis-tests-3_9.yml:16]: One quick enhancement suggestion here, let's follow a standard yaml pattern with below snippet everywhere in CI files!
```
branches:
      - main
      - '0.[0-9]+.[0-9]+'
```

**Review by akash-jain-10 — APPROVED** [review]

**Review by akash-jain-10 — APPROVED** [review]

File: .github/workflows/sync-docs.yml (hunk 1 above)
Comment (Ashok810) [reply .github/workflows/sync-docs.yml]: followed a standard yaml pattern

File: .github/workflows/selenium-noIngestion-tests.yml (hunk 2 above)
Comment (Ashok810) [reply .github/workflows/selenium-noIngestion-tests.yml]: followed a standard yaml pattern

File: .github/workflows/py-tests.yml (hunk 4 above)
Comment (Ashok810) [reply .github/workflows/py-tests.yml]: followed a standard yaml pattern

File: .github/workflows/py-checkstyle.yml (hunk 6 above)
Comment (Ashok810) [reply .github/workflows/py-checkstyle.yml]: followed a standard yaml pattern

File: .github/workflows/maven-build.yml (hunk 8 above)
Comment (Ashok810) [reply .github/workflows/maven-build.yml]: followed a standard yaml pattern

File: .github/workflows/java-checkstyle.yml (hunk 10 above)
Comment (Ashok810) [reply .github/workflows/java-checkstyle.yml]: followed a standard yaml pattern

File: .github/workflows/cypress-integration-tests-postgresql.yml (hunk 12 above)
Comment (Ashok810) [reply .github/workflows/cypress-integration-tests-postgresql.yml]: followed a standard yaml pattern

File: .github/workflows/airflow-apis-tests-3_9.yml (hunk 14 above)
Comment (Ashok810) [reply .github/workflows/airflow-apis-tests-3_9.yml]: followed a standard yaml pattern

File: .github/workflows/cypress-integration-tests-mysql.yml (hunk 13 above)
Comment (Ashok810) [reply .github/workflows/cypress-integration-tests-mysql.yml]: followed a standard yaml pattern

**Pull request merged**